src/ucp/rma/flush.c

```diff
@@ -63,6 +64,28 @@ static int ucp_ep_flush_is_completed(ucp_request_t *req)
     return (req->send.state.uct_comp.count == 0) && req->send.flush.sw_done;
 }
 
+static int uct_ep_flush_is_needed(ucp_ep_h ep, ucp_lane_index_t lane)
+{
+    ucp_worker_h worker = ep->worker;
+    ucp_worker_iface_t *wiface;
+
+    if (!worker->context->config.ext.proto_enable) {
+        return 1;
+    }
+
+    if (ep->flags & UCP_EP_FLAG_INTERNAL) {
```

Comment:
why need different handling for internal ep?

Reply:
We always activate wiface for mem type endpoints.